x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -261,7 +262,7 @@ public void authorizeIndexAction(RequestInfo requestInfo, AuthorizationInfo auth
                 // if the action is a search scroll action, we first authorize that the user can execute the action for some
                 // index and if they cannot, we can fail the request early before we allow the execution of the action and in
                 // turn the shard actions
-                if (SearchScrollAction.NAME.equals(action)) {
+                if (SearchScrollAction.NAME.equals(action) && ((SearchScrollRequest) request).hasLocalIndices()) {
```

Comment:
I would avoid throwing exceptions in this part of the code, it's very brittle, better call the `listener.onFailure` .

Reply:
This is a good catch. Thanks. I have added the exception handling and listener call. It is a pretty basic `try..catch` block. Please let me know if there are better way or convention to do it differently. Thanks.